### PR TITLE
Fix dynamic orientation change while having image view open

### PR DESCRIPTION
**Pull request by dickerpulli**
According to https://github.com/antonKalinin/react-native-image-view/issues/15

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -65,46 +65,51 @@ const SCALE_MAX_MULTIPLIER = 3;
...
-        color: '#FFF',
-        textAlign: 'center',
-    },
-    footer: {
-        position: 'absolute',
-        bottom: 0,
-        left: 0,
-        zIndex: 100,
-    },
+function createStyles(screenDimensions) {
```
Comment (antonKalinin) [inline index.js]: Please use object destructuring here `createStyles({screenWidth, screenHeight})`

File: index.js [hunk 2]
```diff
@@ -126,8 +131,8 @@ const pow2abs = (a: number, b: number): number => Math.pow(Math.abs(a - b), 2);
 
 function getItemLayout(data, index): any {
     return {
```
Comment (antonKalinin) [inline index.js]: `const {screenWidth, screenHeight} = this.state.screenDimensions`

File: index.js [hunk 3]
```diff
@@ -144,29 +149,31 @@ function getDistance(touches: Array<TouchType>): number {
...
     }
 
     return 1;
 }
 
 function calculateInitalTranslate(
     imageWidth: number = 0,
-    imageHeight: number = 0
+    imageHeight: number = 0,
+    screenDimensions
```
Comment (antonKalinin) [inline index.js]: `{screenWidth, screenHeight}`

File: index.js [hunk 4]
```diff
@@ -235,7 +242,11 @@ export default class ImageView extends Component<PropsType> {
         this.renderCounter = 1;
 
         // calculate initial scale and translate for images
-        this.imageInitialParams = props.images.map(getInitalParams);
+        let initialScreenDimensions = {
```
Comment (antonKalinin) [inline index.js]: `const initialScreenDimensions `

Comment (antonKalinin) [reply index.js]: Moreover this pattern: 
```js
{
  screenWidth: Dimensions.get('window').width,
  screenHeight: Dimensions.get('window').height,
}
``` 
is repeating so its better to place it in function like 

```js

const getScreenDimensions () => ({
  screenWidth: Dimensions.get('window').width,
  screenHeight: Dimensions.get('window').height,
})
```

and use it here and in the beggining

File: index.js [hunk 5]
```diff
@@ -333,7 +366,7 @@ export default class ImageView extends Component<PropsType> {
     }
 
     onNextImagesReceived(images: Array<ImageType>, imageIndex: number = 0) {
-        this.imageInitialParams = images.map(getInitalParams);
+        this.imageInitialParams = images.map(getInitalParams.bind(null, this.state.screenDimensions));
```
Comment (antonKalinin) [inline index.js]: Please, retrieve `screenDimensions` from `this.state` in const variable and use it farther in this method

File: index.js [hunk 6]
```diff
@@ -65,46 +65,51 @@ const SCALE_MAX_MULTIPLIER = 3;
...
+      },
+      footer: {
+          position: 'absolute',
+          bottom: 0,
+          left: 0,
+          zIndex: 100,
+      },
+  });
+}
+var styles = createStyles({
```
Comment (antonKalinin) [inline index.js]: `let styles`

**Comment by antonKalinin** [comment]
@dickerpulli Many thanks for this PR. It'g good but needs some improvements, mainly because of using object destructuring in this code before.

File: index.js (hunk 5 above)
Comment (dickerpulli) [reply index.js]: *this.state* is not reachable in const function

**Comment by dickerpulli** [comment]
Code optimized :-)

**Pull request merged**